src/hera/workflows/_mixins.py

```diff
@@ -881,6 +946,38 @@ class TemplateInvocatorSubNodeMixin(BaseMixin):
     when: Optional[str] = None
     with_sequence: Optional[Sequence] = None
 
+    def with_(self, **kwargs: dict) -> TemplateInvocatorSubNodeMixin:
+        """Update the created Step/Task object with the attributes you want to pass.
+
+        Used for setting the attributes such as name, when, or with_param. Setting with_param
+        and with_items triggers a recalculation of arguments.
+        """
+        if (
+            self.template is None
+            or not hasattr(self.template, "directly_callable")
```

Comment:
Why not checking `isinstance(self.template, CallableTemplateMixin)` which is equiv to these two conditions